firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataConverter.java

```diff
@@ -194,25 +192,30 @@ public FieldValue parseQueryValue(Object input) {
 
   /** Converts a POJO to native types and then parses it into model types. */
   private FieldValue convertAndParseData(Object input, ParseContext context) {
-    UserData.Source source = context.getDataSource();
+    Object converted = CustomClassMapper.convertToPlainJavaTypes(input);
+    return parseData(converted, context);
```

Comment:
Should this be `convertAndParseFieldData()` now?

Reply:
I didn't want to do that, since it's used by convertAndParseDocumentData(), but your suggestion to just inline the implementation was reasonable...  So done.  Though this did add a net 4 lines of code. :-(